test/common/pixel-tests

```diff
@@ -33,6 +38,20 @@ cmd_status() {
 }
 
 cmd_push() {
+    # Check if the current branch is behind the project's main and has newer pixel-tests, in this case
+    # the user has to rebase to not run into pixel-tests conflicts. There is no
+    # elegant way figuring out the real "origin" of the cockpit project so we
+    # discover the project's repository name and hardcode the upstream url.
+    project=$(git remote -v | awk '/^origin/ { print $2; exit }' | xargs basename)
+    git fetch https://github.com/cockpit-project/${project} main
+
+    cur_sha=$(get_pixel_reference "FETCH_HEAD")
+    main_sha=$(get_pixel_reference "HEAD")
+    if [[ "$cur_sha" != "$main_sha" ]]; then
```

Comment:
This is too restrictive, I think. This also prevents running `pixel-tests push` a second time in the same PR.

We need to compare the SHA on main with the one on the common ancestor of main and the current branch.  If they are different, then changing test/reference on the current branch will lead to a conflict.

Reply:
`git merge-base` can find the common ancestor.